### ISSUE TITLE
CreateMotifMatrix

### ISSUE THREAD

**Issue by Brawni** [timoast/signac#130]
Hi Tim!

Thanks again for your package and your support!
Im having troubles generating a MotifsMatrix using a PWMatrixlist which i assembled from PWMs from the cisdb repository (http://cisbp.ccbr.utoronto.ca/index.php).
I Run into the following error:
`
motif.matrix = CreateMotifMatrix(features = StringToGRanges(rownames(signac_obj), sep = c(":", "-")), pwm = pwm, genome = 'hg38', sep = c(":", "-"))

Error in pwmType(pwm) : 
  Can't determine format of PWM -- should be numeric frequency summing to 1 or log or log2 odds ratio

I checked if frequencies of each PWM sums to 1 and in facts some of them don't sum up exactly to 1 but are like 0.999 or 1.001..

pwm           
PWMatrixList of length 4351
names(4351): M00217_2.00 M00218_2.00 M00219_2.00 ... M11437_2.00 M11491_2.00

table (unlist( lapply( lapply( pwm@listData, colSums), function (x) all (x == 1))))
FALSE  TRUE 
 4003   348 
`
Any suggestions on how i can fix this?

Much appreciated!

**Comment by timoast** [comment]
Hi, I think you need to normalize the PWMs so that the columns sum to exactly one. The chromVARmotifs package has some curated data from cisdb where they've already performed some normalization, maybe this could help? https://github.com/GreenleafLab/chromVARmotifs

If you need to use the full set of PWMs that you have now you could just divide each column by the column sum so that they sum to exactly one, for example:

```r
# example matrix from cisdb
pwm <- t(read.table("~/Downloads/Homo_sapiens_2020_04_12_9_27_pm/pwms_all_motifs/M00008_2.00.txt", header = TRUE, row.names = 1))
pwm.norm <- pwm / colSums(pwm)

# to apply to a whole PWMatrixList
for (i in seq_along(pwm)) {
  pwm[[i]]@profileMatrix <- pwm[[i]]@profileMatrix / colSums(pwm[[i]]@profileMatrix)
}

```

**Comment by Brawni** [comment]
the chromVARmotifs was exactly what i was looking for!! 
I also tried normalising the pwms but without success. In facts i think I stumbled upon a rather annoying problem on how R deals with arithmetics:
colSums(pwm[[1]]@profileMatrix)
1  2  3  4  5  6  7  8  9 10 11 
 1  1  1  1  1  1  1  1  1  1  1 
colSums(pwm[[1]]@profileMatrix) == 1
  1     2     3     4     5     6     7     8     9    10    11 
FALSE  TRUE FALSE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE  TRUE 

As far as i understood it has to do with the fact that sums of floating point won't ever match exactly 1 integer. I wonder if that's why i get the error when trying to input the PWMatrixlist. 

Anyway will use the chromVARmotifs pkg so all good!

Thanks!!
